src/main.rs

```diff
@@ -218,45 +320,150 @@ async fn apply(
         }
     };
 
+    let all_eips = eip_api.list(&ListParams::default()).await?.items;
+    let eip = all_eips
+        .into_iter()
+        .find(|eip| &eip.spec.pod_name == pod_name)
+        .ok_or_else(|| Error::NoEipResourceWithThatPodName(pod_name.to_owned()))?;
```

Comment:
Hm, is it a problem if an Eip resource is found that doesn't have an associated pod? It might have gotten destroyed in an upgrade, or we might be pre-allocating it...

Reply:
That's the expected case. The Eip controller is only responsible for allocating the AWS EIP object. The Pod controller is responsible for looking for a K8S Eip resource with a matching podName, getting the allocation ID out of it, and associating it with the pod ENI. If the pod controller can't find the Eip or the Eip doesn't have an allocation ID, it just tries again a few seconds later.